gradio/layouts.py

```diff
@@ -167,15 +175,20 @@ class TabItem(BlockContext):
     def __init__(
         self,
         label: str,
+        *,
         id: Optional[int | str] = None,
+        visible: bool = True,
```

Comment:
Why do we need `visible` for the TabItem, isn't that what `selected` in the `Tabs` class is for?

Reply:
Hmm on second thought, I think I can see a case for needing to show an additional tab based on input defined elsewhere in the app!